src/effects/Saturation.cpp

```diff
@@ -33,13 +33,13 @@
 using namespace openshot;
 
 /// Blank constructor, useful when using Json to load the effect properties
-Saturation::Saturation() : saturation(1.0) {
+Saturation::Saturation() : saturation(1.0), saturation_R(1.0), saturation_G(1.0), saturation_B(1.0) {
 	// Init effect properties
 	init_effect_details();
 }
 
 // Default constructor
-Saturation::Saturation(Keyframe new_saturation) : saturation(new_saturation)
+Saturation::Saturation(Keyframe new_saturation, Keyframe new_saturation_R, Keyframe new_saturation_G, Keyframe new_saturation_B) : saturation(new_saturation), saturation_R(new_saturation_R), saturation_G(new_saturation_G), saturation_B(new_saturation_B)
```

Comment:
In addition to, IMHO, changing these parameter names (see above), please wrap this for readability. At _least_ a line break between the parameters and the initializers. Good editors will auto-wrap and format everything nicely but not _all_ editors will, and horizontal scrolling in GitHub code listings is one of the circles of hell.

Reply:
Also, is it worth providing any other constructors?
* A 1-parameter version that only takes a `saturation` value, and sets the other three `1.0`.
* A 2-parameter version that sets `saturation` to one value, and all three channel sats to another? (Probably this one's a bad idea.)
* A 3-parameter version that takes values for all three channel saturations, and sets `saturation` to 1.0?

Definitely doesn't make sense to do _all_ of those, but perhaps one or two, if they might be useful?

Reply:
> In addition to, IMHO, changing these parameter names (see above), please wrap this for readability. At _least_ a line break between the parameters and the initializers. Good editors will auto-wrap and format everything nicely but not _all_ editors will, and horizontal scrolling in GitHub code listings is one of the circles of hell.

Fixed by 3a987eb044e0d755b4a1ae321fd816572e246e27.

Reply:
> Also, is it worth providing any other constructors?
> 
> * A 1-parameter version that only takes a `saturation` value, and sets the other three `1.0`?
> * A 2-parameter version that sets `saturation` to one value, and all three channel sats to another? (Probably this one's a bad idea.)
> * A 3-parameter version that takes values for all three channel saturations, and sets `saturation` to 1.0?
> 
> Definitely doesn't make sense to do _all_ of those, but perhaps one or two, if they might be useful?

Sounds like premature optimization, as there does not exist any code actually invoking those, and it is most simple to add any of those later in a separate PR once needed.